### ISSUE TITLE
 SSL verification error at depth 1: self signed certificate in certificate chain

### ISSUE THREAD

**Issue by yuniq-neko** [rubygems/rubygems#2791]
Hi there, I'm trying to install Jekyll but having a issue with Gem where I need to use a self signed certificate to access HTTPS; as my business firewall does SSL Inspection, however for the life of me... cannot work out how to install the certificate into Rubygems. When performing `gem update --system` for example... itll spew out the following: 
```
ERROR:  SSL verification error at depth 1: self signed certificate in certificate chain (19)
ERROR:  Root certificate is not trusted (/C=AU/ST=Victoria/O=###/OU=###/CN=###CA/emailAddress=###)
ERROR:  While executing gem ... (Gem::RemoteFetcher::FetchError)
    SSL_connect returned=1 errno=0 state=error: certificate verify failed (self signed certificate in certificate chain) (https://api.rubygems.org/specs.4.8.gz)
```
I have obfuscated the text after 'not trusted' message due to it containing sensitive text. 
Running Windows 10 Build 1903 with following gem version:
```
$ gem env version
2.7.6.2
```
The most I have done so far is export the certificate from Windows Certificate Manager as `Base-64 encoded X.509 (.CER)` and place it in `C:\Ruby25-x64\lib\ruby\2.5.0\rubygems\ssl_certs`... I also tried renaming it to `AddTrustExternalCARoot.pem`... still get the error... I do also have the original certificate in its `.crt` form.

TLDR: How does one import a root certificate into Gem on Windows?

**Comment by indirect** [comment]
The certificates corresponding to `rubygems.org` are hardcoded, and there is no supported way to replace them, sorry. If you are connecting to a server that is not rubygems.org, we recommend obtaining valid SSL certificates for that server and using those instead. Other possible workarounds might include setting up a caching or proxy server (like Gemstash), or downloading the .gem files via some other means and then installing the downloaded files using `gem install /path/to/downloaded.gem`.

**Comment by yuniq-neko** [comment]
ok; I've used a workaround for now being disable SSL decryption for rubygems.org and *.rubygems.org... Seems to have worked so far.

**Comment by flovlad** [comment]
> ok; I've used a workaround for now being disable SSL decryption for rubygems.org and *.rubygems.org... Seems to have worked so far.

What was that workaround exactly?

For me the working solution was to download the root CA from  https://rubygems.org/
and then in my Dockerfile (FROM ubuntu:bionic) to:

openssl x509 -inform DER -in /tmp/ruby-root-ca.cer -out /tmp/ruby-root-ca.crt
cp /tmp/ruby-root-ca.crt /usr/local/share/ca-certificates/
update-ca-certificates
